app/controllers/users/reset_passwords_controller.rb

```diff
@@ -83,6 +100,11 @@ def handle_unsuccessful_password_reset(result)
       render :edit
     end
 
+    def update_user
+      resource.update(confirmed_at: Time.current) unless resource.confirmed?
```

Comment:
not sure how you feel about removing Devise, I have been trying to do it when I can since I think at the level of customization we have it is more confusing than helpful.

If you are interested in moving away from Devise, removing references to `resource` would help :)